diff --git a/ydb/core/grpc_services/rpc_alter_table.cpp b/ydb/core/grpc_services/table/rpc_alter_table.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_alter_table.cpp
rename to ydb/core/grpc_services/table/rpc_alter_table.cpp
diff --git a/ydb/core/grpc_services/rpc_begin_transaction.cpp b/ydb/core/grpc_services/table/rpc_begin_transaction.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_begin_transaction.cpp
rename to ydb/core/grpc_services/table/rpc_begin_transaction.cpp
diff --git a/ydb/core/grpc_services/rpc_commit_transaction.cpp b/ydb/core/grpc_services/table/rpc_commit_transaction.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_commit_transaction.cpp
rename to ydb/core/grpc_services/table/rpc_commit_transaction.cpp
diff --git a/ydb/core/grpc_services/rpc_copy_table.cpp b/ydb/core/grpc_services/table/rpc_copy_table.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_copy_table.cpp
rename to ydb/core/grpc_services/table/rpc_copy_table.cpp
diff --git a/ydb/core/grpc_services/rpc_copy_tables.cpp b/ydb/core/grpc_services/table/rpc_copy_tables.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_copy_tables.cpp
rename to ydb/core/grpc_services/table/rpc_copy_tables.cpp
diff --git a/ydb/core/grpc_services/rpc_create_session.cpp b/ydb/core/grpc_services/table/rpc_create_session.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_create_session.cpp
rename to ydb/core/grpc_services/table/rpc_create_session.cpp
diff --git a/ydb/core/grpc_services/rpc_create_table.cpp b/ydb/core/grpc_services/table/rpc_create_table.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_create_table.cpp
rename to ydb/core/grpc_services/table/rpc_create_table.cpp
diff --git a/ydb/core/grpc_services/rpc_delete_session.cpp b/ydb/core/grpc_services/table/rpc_delete_session.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_delete_session.cpp
rename to ydb/core/grpc_services/table/rpc_delete_session.cpp
diff --git a/ydb/core/grpc_services/rpc_describe_table.cpp b/ydb/core/grpc_services/table/rpc_describe_table.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_describe_table.cpp
rename to ydb/core/grpc_services/table/rpc_describe_table.cpp
diff --git a/ydb/core/grpc_services/rpc_describe_table_options.cpp b/ydb/core/grpc_services/table/rpc_describe_table_options.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_describe_table_options.cpp
rename to ydb/core/grpc_services/table/rpc_describe_table_options.cpp
diff --git a/ydb/core/grpc_services/rpc_drop_table.cpp b/ydb/core/grpc_services/table/rpc_drop_table.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_drop_table.cpp
rename to ydb/core/grpc_services/table/rpc_drop_table.cpp
diff --git a/ydb/core/grpc_services/rpc_execute_data_query.cpp b/ydb/core/grpc_services/table/rpc_execute_data_query.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_execute_data_query.cpp
rename to ydb/core/grpc_services/table/rpc_execute_data_query.cpp
diff --git a/ydb/core/grpc_services/rpc_execute_scheme_query.cpp b/ydb/core/grpc_services/table/rpc_execute_scheme_query.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_execute_scheme_query.cpp
rename to ydb/core/grpc_services/table/rpc_execute_scheme_query.cpp
diff --git a/ydb/core/grpc_services/rpc_explain_data_query.cpp b/ydb/core/grpc_services/table/rpc_explain_data_query.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_explain_data_query.cpp
rename to ydb/core/grpc_services/table/rpc_explain_data_query.cpp
diff --git a/ydb/core/grpc_services/rpc_explain_yql_script.cpp b/ydb/core/grpc_services/table/rpc_explain_yql_script.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_explain_yql_script.cpp
rename to ydb/core/grpc_services/table/rpc_explain_yql_script.cpp
diff --git a/ydb/core/grpc_services/rpc_keep_alive.cpp b/ydb/core/grpc_services/table/rpc_keep_alive.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_keep_alive.cpp
rename to ydb/core/grpc_services/table/rpc_keep_alive.cpp
diff --git a/ydb/core/grpc_services/rpc_load_rows.cpp b/ydb/core/grpc_services/table/rpc_load_rows.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_load_rows.cpp
rename to ydb/core/grpc_services/table/rpc_load_rows.cpp
diff --git a/ydb/core/grpc_services/rpc_prepare_data_query.cpp b/ydb/core/grpc_services/table/rpc_prepare_data_query.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_prepare_data_query.cpp
rename to ydb/core/grpc_services/table/rpc_prepare_data_query.cpp
diff --git a/ydb/core/grpc_services/rpc_read_rows.cpp b/ydb/core/grpc_services/table/rpc_read_rows.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_read_rows.cpp
rename to ydb/core/grpc_services/table/rpc_read_rows.cpp
diff --git a/ydb/core/grpc_services/rpc_read_table.cpp b/ydb/core/grpc_services/table/rpc_read_table.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_read_table.cpp
rename to ydb/core/grpc_services/table/rpc_read_table.cpp
diff --git a/ydb/core/grpc_services/rpc_rename_tables.cpp b/ydb/core/grpc_services/table/rpc_rename_tables.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_rename_tables.cpp
rename to ydb/core/grpc_services/table/rpc_rename_tables.cpp
diff --git a/ydb/core/grpc_services/rpc_rollback_transaction.cpp b/ydb/core/grpc_services/table/rpc_rollback_transaction.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_rollback_transaction.cpp
rename to ydb/core/grpc_services/table/rpc_rollback_transaction.cpp
diff --git a/ydb/core/grpc_services/rpc_stream_execute_scan_query.cpp b/ydb/core/grpc_services/table/rpc_stream_execute_scan_query.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_stream_execute_scan_query.cpp
rename to ydb/core/grpc_services/table/rpc_stream_execute_scan_query.cpp
diff --git a/ydb/core/grpc_services/rpc_stream_execute_yql_script.cpp b/ydb/core/grpc_services/table/rpc_stream_execute_yql_script.cpp
similarity index 100%
rename from ydb/core/grpc_services/rpc_stream_execute_yql_script.cpp
rename to ydb/core/grpc_services/table/rpc_stream_execute_yql_script.cpp
diff --git a/ydb/core/grpc_services/ya.make b/ydb/core/grpc_services/ya.make
index d6e169c22a9..d55719ce8ad 100644
--- a/ydb/core/grpc_services/ya.make
+++ b/ydb/core/grpc_services/ya.make
@@ -12,32 +12,32 @@ SRCS(
     operation_helpers.cpp
     resolve_local_db_table.cpp
     rpc_alter_coordination_node.cpp
-    rpc_alter_table.cpp
-    rpc_begin_transaction.cpp
+    table/rpc_alter_table.cpp
+    table/rpc_begin_transaction.cpp
     rpc_calls.cpp
     rpc_cancel_operation.cpp
     rpc_cms.cpp
-    rpc_commit_transaction.cpp
+    table/rpc_commit_transaction.cpp
     rpc_dynamic_config.cpp
-    rpc_copy_table.cpp
-    rpc_copy_tables.cpp
+    table/rpc_copy_table.cpp
+    table/rpc_copy_tables.cpp
     rpc_export.cpp
     rpc_create_coordination_node.cpp
-    rpc_create_session.cpp
-    rpc_create_table.cpp
-    rpc_delete_session.cpp
+    table/rpc_create_session.cpp
+    table/rpc_create_table.cpp
+    table/rpc_delete_session.cpp
     rpc_describe_coordination_node.cpp
     rpc_describe_path.cpp
-    rpc_describe_table.cpp
-    rpc_describe_table_options.cpp
+    table/rpc_describe_table.cpp
+    table/rpc_describe_table_options.cpp
     rpc_drop_coordination_node.cpp
-    rpc_drop_table.cpp
+    table/rpc_drop_table.cpp
     rpc_discovery.cpp
-    rpc_execute_data_query.cpp
-    rpc_execute_scheme_query.cpp
-    rpc_execute_yql_script.cpp
-    rpc_explain_yql_script.cpp
-    rpc_explain_data_query.cpp
+    table/rpc_execute_data_query.cpp
+    table/rpc_execute_scheme_query.cpp
+    table/rpc_execute_yql_script.cpp
+    table/rpc_explain_yql_script.cpp
+    table/rpc_explain_data_query.cpp
     rpc_forget_operation.cpp
     rpc_fq_internal.cpp
     rpc_fq.cpp
@@ -51,24 +51,24 @@ SRCS(
     rpc_kqp_base.cpp
     rpc_list_operations.cpp
     rpc_login.cpp
-    rpc_load_rows.cpp
+    table/rpc_load_rows.cpp
     rpc_log_store.cpp
     rpc_long_tx.cpp
     rpc_node_registration.cpp
     rpc_make_directory.cpp
     rpc_modify_permissions.cpp
     rpc_monitoring.cpp
-    rpc_prepare_data_query.cpp
+    table/rpc_prepare_data_query.cpp
     rpc_rate_limiter_api.cpp
     rpc_read_columns.cpp
-    rpc_read_table.cpp
-    rpc_read_rows.cpp
+    table/rpc_read_table.cpp
+    table/rpc_read_rows.cpp
     rpc_remove_directory.cpp
-    rpc_rename_tables.cpp
-    rpc_rollback_transaction.cpp
+    table/rpc_rename_tables.cpp
+    table/rpc_rollback_transaction.cpp
     rpc_scheme_base.cpp
-    rpc_stream_execute_scan_query.cpp
-    rpc_stream_execute_yql_script.cpp
+    table/rpc_stream_execute_scan_query.cpp
+    table/rpc_stream_execute_yql_script.cpp
     rpc_whoami.cpp
     table_settings.cpp
 
